dpctl/__init__.py

```diff
@@ -100,6 +101,7 @@
     "SyclKernelInvalidRangeError",
     "SyclKernelSubmitError",
     "SyclQueueCreationError",
+    "WorkGroupMemory",
```

Comment:
Should we expose this in the main `dpctl` namespace? The `LocalAccessor` PR chose not to expose it, and none of the other kernel arguments are exposed either.

This is mostly out of curiosity, if it greatly hurts usability, I don't see a problem.

Reply:
The way the `LocalAccessor` PR exposes (or rather not exposes) the feature through `_sycl_queue` indicates (due to the `_` in the name) to users that this feature is not supposed to be used in applications. 

I don't think there is a reason we should hide this feature (or `LocalAccessor` FWIW) from users, they are user-available features in SYCL. 

For our own use case, it wouldn't make too much of a difference, so I can stop exposing this in the namespace if you prefer to "hide" this feature.

Reply:
I agree in general, I guess it is only regrettable that there isn't a more organized/obvious way to indicate to a user that these are kernel argument types, like a namespace or something.

But I don't think it's too important.

The only thing I think should be added (with respect to this) is specific mention that this class is intended to be used as a kernel argument type, same as in `LocalAccessor`, which I agree, should also be exposed.

Reply:
Yeah, I see your point about marking them as kernel arguments.

For now, I've added a sentence to the documentation of the class to clarify its intent.

The next contribution I'm planning to bring based on this one (support for `raw_kernel_arg`) also falls into this category, so maybe after this, `LocalAccessor` and the `raw_kernel_arg` have landed, we can find a way to mark them as kernel arguments.